sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample_MockClient.md

```diff
@@ -0,0 +1,88 @@
+
+```C# Snippet:FormRecognizerCreateMocks
+var fakeModelId = Guid.NewGuid().ToString();
+var fakeOperationId = $"{fakeModelId}/analyzeResults/{Guid.NewGuid()}";
+var mockClient = new Mock<FormRecognizerClient>();
+var mockOperation = new Mock<RecognizeCustomFormsOperation>(fakeOperationId, mockClient.Object);
+```
+
+Then, set up the client methods that will be executed. In this case, we will call the `StartRecognizeCustomFormsFromUriAsync` method.
+
+```C# Snippet:FormRecognizerSetUpClientMock
+var fakeDocumentUri = new Uri("https://fake.document.uri");
+
+mockClient.Setup(c => c.StartRecognizeCustomFormsFromUriAsync(fakeModelId, fakeDocumentUri,
+    It.IsAny<RecognizeCustomFormsOptions>(), It.IsAny<CancellationToken>()))
+    .Returns(Task.FromResult(mockOperation.Object));
+```
+
+If you're mocking an operation object, you will also need to set up the methods that will be called from it. In this sample, we only need to set `WaitForCompletionAsync`.
+
+```C# Snippet:FormRecognizerSetUpOperationMock
+var labelDataBox = FormRecognizerModelFactory.FieldBoundingBox(new List<PointF>()
+    { new PointF(1f, 1f), new PointF(2f, 1f), new PointF(2f, 2f), new PointF(1f, 2f) });
+var labelData = FormRecognizerModelFactory.FieldData(labelDataBox, 1, "Total price:", new List<FormElement>());
+
+var valueDataBox = FormRecognizerModelFactory.FieldBoundingBox(new List<PointF>()
+    { new PointF(4f, 1f), new PointF(5f, 1f), new PointF(5f, 2f), new PointF(4f, 2f) });
+var valueData = FormRecognizerModelFactory.FieldData(valueDataBox, 1, "$150.00", new List<FormElement>());
+
+var fieldValue = FormRecognizerModelFactory.FieldValueWithFloatValueType(150f);
+
+var formField = FormRecognizerModelFactory.FormField("totalPrice", labelData, valueData, fieldValue, 0.85f);
+var formPage = FormRecognizerModelFactory.FormPage(1, 8.5f, 11f, 0f, LengthUnit.Inch, new List<FormLine>(), new List<FormTable>());
+
+var pageRange = FormRecognizerModelFactory.FormPageRange(1, 1);
+var recognizedForm = FormRecognizerModelFactory.RecognizedForm("groceries", pageRange,
+    new Dictionary<string, FormField>() { { "totalPrice", formField } },
+    new List<FormPage>() { formPage });
+var recognizedFormCollection = FormRecognizerModelFactory.RecognizedFormCollection(new List<RecognizedForm>() { recognizedForm });
+
+Response<RecognizedFormCollection> operationResponse = Response.FromValue(recognizedFormCollection, Mock.Of<Response>());
+
+mockOperation.Setup(op => op.WaitForCompletionAsync(It.IsAny<CancellationToken>()))
+    .Returns(new ValueTask<Response<RecognizedFormCollection>>(operationResponse));
+```
+
+To keep the setup simple, we suggest that you only add the properties required by your tests when building your models. In this case, we only needed the `FormType`, the field's `Confidence`, and the field's `Value`, while the other properties could be set to `null`, `default`, or empty.
```

Comment:
Considered following my own suggestion and making this setup simpler, but I wanted to illustrate a more robust mocking in case someone needs it.